forge/src/main/java/me/shedaniel/architectury/event/forge/EventHandlerImplClient.java

```diff
@@ -115,59 +116,69 @@ public static void event(ClientChatReceivedEvent event) {
         if (process.getResult() == InteractionResult.FAIL)
             event.setCanceled(true);
     }
-    
+
     @SubscribeEvent
     public static void event(WorldEvent.Save event) {
         if (event.getWorld() instanceof ClientLevel) {
             ClientLevel world = (ClientLevel) event.getWorld();
             ClientLifecycleEvent.CLIENT_WORLD_LOAD.invoker().act(world);
         }
     }
-    
+
+    @SubscribeEvent
+    public static void event(GuiOpenEvent event) {
+        InteractionResultHolder<Screen> result = GuiEvent.SET_SCREEN.invoker().modifyScreen(event.getGui());
+        if (result.getResult() == InteractionResult.FAIL) {
+            event.setCanceled(true);
+            return;
+        }
+        event.setGui(result.getObject());
```

Comment:
Only apply if SUCCESS

Reply:
What should the analogue be on fabric? Should we also only change on SUCCESS?

Reply:
Yep